pkgs/development/cuda-modules/cuda/overrides.nix

```diff
@@ -43,9 +47,42 @@ attrsets.filterAttrs (attr: _: (builtins.hasAttr attr prev)) {
   );
 
   cuda_compat = prev.cuda_compat.overrideAttrs (
-    prevAttrs: {
-      env.autoPatchelfIgnoreMissingDeps =
-        prevAttrs.env.autoPatchelfIgnoreMissingDeps + " libnvrm_gpu.so libnvrm_mem.so";
+    prevAttrs: rec {
+      env.autoPatchelfIgnoreMissingDeps = builtins.concatStringsSep " " (
+        [
+          prevAttrs.env.autoPatchelfIgnoreMissingDeps
+          "libnvrm_gpu.so libnvrm_mem.so"
+        ]
+        ++ libcudaExtraNeeded
+      );
+
+      # Append the jetpack's libnvrm_{gpu,mem}.so impure location with the lowest priority.
+      # This way we trade safety for the out-of-the box experience on jetsons.
+      appendRunpaths = [
+        "${stdenv.cc.cc.lib}/lib"
+      ] ++ prevAttrs.appendRunpaths ++ ["/usr/lib/aarch64-linux-gnu/tegra/"];
+
+      libcudaExtraNeeded = [
+        "libnvos.so"
+        "libnvsocsys.so"
+        "libnvrm_sync.so"
+        "libnvos.so"
+        "libnvsciipc.so"
+        "libnvsocsys.so"
+        "libnvrm_chip.so"
+        "libnvrm_host1x.so"
+        "libstdc++.so"
+      ];
```

Comment:
Quoting the message from matrix:

> I'm wondering if we could somehow move this to nixglhost (e.g. as LD_PRELOAD) or maybe we could write and link a library that would scan /usr/lib/aarch-blahblaah/tegra and dlopen stuff in the correct order.

...presumably, this list might change at time, i.e. our current nixpkgs revision might not be compatible with the future jetpacks